x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -434,7 +431,7 @@ void runCompute(CancellableTask task, ComputeContext context, PhysicalPlan plan,
         driverRunner.executeDrivers(
             task,
             drivers,
-            transportService.getThreadPool().executor(ESQL_WORKER_THREAD_POOL_NAME),
+            transportService.getThreadPool().executor(ThreadPool.Names.SEARCH_WORKER),
```

Comment:
This specific bit requires additional thinking: I see that the esql worker has a bounded queue. Is this thread pool ever expected to reject requests? 

In _search, the search thread pool is the entry point, and for one request that is parallelized, you will have the search "coordination" thread blocking and waiting for all workers that it parallelizes across to complete their operation. We can reliably say that the search thread pool is the one applying back pressure. Once something entered the search thread pool, we need to make sure that the workers execute it and never reject anything. That is why we ended up with an unbounded queue for the search worker thread pool. This does build on how inter-segment concurrency works in Lucene, in that it is not async, and the coordinator blocks and waits.

I believe that esql works differently, and that the two thread pools are rather separate and not really dependent on each other. I believe that esql would rather prefer a bounded queue for the worker side (please correct me if I am wrong) but that would break the assumptions we make on the Lucene inter-segment concurrency side of things.

Could you help me understand potential bits that I missed here, and the full interaction between esql and worker thread pools in esql please?

Reply:
Thanks, Luca! I think you've raised a great point here. ESQL operates similarly to the search API in terms of coordination, with one important difference: the esql threadpool is never blocked by the esql_worker. This means that if we replace the esql and esql_worker with the search and search_worker threadpools, we might not see the rejections that we should see before this change. That could potentially lead to OOM.

1. I think we can replace the esql threadpool with the search threadpool, while leaving the esql_worker unchanged for now.

2. As for the esql_worker threadpool, we can leave it as is for now and account it into our autoscaling, or alternatively, block ESQL data-node requests to mimic the behavior in the search API. What do you think?

Reply:
I think I would replace the esql thread pool with the search one to start with. That seems safe to do. I'd leave search worker and esql worker as-is for now. What to do with esql worker needs further discussion, as well as how auto-scaling takes it into account.

Reply:
Yes, I have updated this PR to replace `esql` by `search` threadpool only in https://github.com/elastic/elasticsearch/pull/106050/commits/87db43e8082618d021e9b87688d5cc7efd75bbac.